src/Symfony/Component/Cache/Adapter/ArrayAdapter.php

```diff
@@ -177,4 +172,16 @@ private function validateKey($key)
 
         return $key;
     }
+
+    private function createGenerator(array $keys)
```

Comment:
@nicolas-grekas this is about making some logic running before the creation of the generator. If you inline the method, all the code is part of the generator and so runs only when you start iterating. Here, the code before the private method is not part of the generator logic.